doc/reference/networking/openthread_integration.rst

```diff
@@ -0,0 +1,135 @@
+.. _openthread_integration:
+
+OpenThread Integration
+######################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+TODO
+
+File system and shim layer
+**************************
+TODO
+
+Threads
+*******
+TODO
+
+Traffic flow
+************
+TODO
+
+RX path
+*******
+An application typically consists of one or more :ref:`threads <threads_v2>`
+that execute the application logic. When using the
+:ref:`BSD socket API <bsd_sockets_interface>`, the following things will
+happen.
+
+.. figure:: zephyr_netstack_openthread-rx_sequence.svg
+    :alt: OpenThread Application RX data flow
+    :figclass: align-center
+
+    OpenThread Application RX data flow
+
+Data receiving (RX) - TODO: expand, remove/rephrase copy/paste from the zephyrs RX page.
+-------------------
+
+1. A network data packet is received by a nRF5 IEEE 802.15.4 Radio Driver.
+
+2. The device driver allocates enough network buffers to store the received
+   data. The network packet is placed in the proper RX queue (implemented by
+   :ref:`k_fifo <fifos_v2>`). By default there is only one receive queue in
+   the system, but it is possible to have up to 8 receive queues.
+   These queues will process incoming packets with different priority.
+   See :ref:`traffic-class-support` for more details. The receive queues also
+   act as a way to separate the data processing pipeline (bottom-half) as
+   the device driver is running in an interrupt context and it must do its
+   processing as fast as possible.
+
+3. The 802154 RX Radio driver thread does the upper-half processing of the
+   received IEEE 802.15.4 radio frame. As a result it puts a work item
+   with the net_recv_data() to have the frame processed.
+
+4. The work queue thread rx_workq calls the openthread_recv() which inserts
+   the frame in the rx_pkt_fifo and returns the NET_OK.
+
+5. The OpenThread thread gets a frame from the FIFO and consumes it.
+   It also handles IP header compression, reassembly of fragmented traffic.
```

Comment:
Not sure if "being consumed" is here or 4 or maybe 6. Packet was removed from zephyr fifo in 4 by the recv function but it is still in use and will be freed after openthread_process.

Reply:
I agree. This is the only thing left to do here when it comes to obvious missing information.
I would like to mark places in which net_pkts are allocated and unreferenced.